modin/engines/base/frame/data.py

```diff
@@ -1628,9 +1633,9 @@ def broadcast_apply_full_axis(
             None,
             dtypes,
         )
-        if new_index is not None:
+        if new_axes[0] is not None:
```

Comment:
Since we compute new variable `new_axis` above based on `new_index` and `new_columns` I would prefer using `new_axis`.

Reply:
This change will impact performance. `new_index` indicates that there will be a mismatch, so external metadata should be propagated into internal. In your changes, the external is always broadcast to the partitions because `new_axes[0]` will always be `not None`. I think it should be reverted and if we want to change this discuss in a separate issue or new PR.

Reply:
Okay, I see. This should be reverted.